openfisca_core/holders.py

```diff
@@ -6,41 +6,57 @@
 
 import numpy as np
 import psutil
+from typing import TYPE_CHECKING, List, Union, Any, Optional, Dict
 
 from openfisca_core import periods
 from openfisca_core.commons import empty_clone
 from openfisca_core.data_storage import InMemoryStorage, OnDiskStorage
 from openfisca_core.errors import PeriodMismatchError
 from openfisca_core.indexed_enums import Enum
-from openfisca_core.periods import MONTH, YEAR, ETERNITY
+from openfisca_core.periods import MONTH, YEAR, ETERNITY, Period
 from openfisca_core.tools import eval_expression
 
+
+if TYPE_CHECKING:
+    from openfisca_core.populations import Population
+    from openfisca_core.variables import Variable
+    from openfisca_core.simulations import Simulation
+
+
 log = logging.getLogger(__name__)
 
 
+#TODO change with a more specific type
+ArrayLike = Any
+
+
 class Holder(object):
     """
         A holder keeps tracks of a variable values after they have been calculated, or set as an input.
     """
 
-    def __init__(self, variable, population):
-        self.population = population
-        self.variable = variable
-        self.simulation = population.simulation
-        self._memory_storage = InMemoryStorage(is_eternal = (self.variable.definition_period == ETERNITY))
+    def __init__(self, variable:'Variable', population:'Population'):
```

Comment:
Flake won't like the lack of spaces. 

Also you can forward-reference them no?